src/js/components/modals/ImageViewerModal.js

```diff
@@ -38,9 +38,24 @@ class ImageViewerModal extends React.Component {
     }
   }
 
+  componentWillUpdate(nextProps, {isLoadingImage}) {
+    let {images, selectedImage} = this.props;
+    let nextImages = nextProps.images;
+    let nextSelectedImage = nextProps.selectedImage;
+    // Two images in a row are the same, img.onLoad will never be called, so
+    // here we are updating manually
+    if (isLoadingImage &&
+        images[selectedImage] === nextImages[nextSelectedImage]) {
+      this.handleLoadingImageChange(false);
```

Comment:
the react docs stress that we shouldn't use `setState` inside of `componentWillUpdate` and to instead do it in `componentWillReceiveProps`. is this possible?